chanbackup/single.go

```diff
@@ -329,6 +419,56 @@ func (s *Single) Serialize(w io.Writer) error {
 		}
 	}
 
+	// Encode version enum and hasCloseTx flag to version byte.
+	version := s.Version.Encode(s.CloseTxInputs.IsSome())
+
+	// Serialize CloseTxInputs if it is provided. Fill err if it fails.
+	var err error
+	s.CloseTxInputs.WhenSome(func(inputs CloseTxInputs) {
+		err = inputs.CommitTx.Serialize(&singleBytes)
+		if err != nil {
+			return
+		}
+
+		err = lnwire.WriteElements(
+			&singleBytes,
+			uint16(len(inputs.CommitSig)), inputs.CommitSig,
+		)
+		if err != nil {
+			return
+		}
+
```

Comment:
nit: could do `!s.Version.IsTaproot()` to save some indentations here

Reply:
Done. Also added a comment to explain what happens after the check:

```go
                if !s.Version.IsTaproot() {
                        return
                }

                // Write fields needed for taproot channels.
                err = lnwire.WriteElements(
                        &singleBytes, inputs.CommitHeight,
                )
                if err != nil {
                        return
                }
                ...
```